### DISCUSSION TITLE
mobx6 and makeAutoObservable produce strict mode warnings

### DISCUSSION THREAD

**Discussion by FalkNisius** [mobxjs/mobx#2901] [General] [answered]
If I use makeAutoObservable, I assume no more attention of changing object properties is needed.
Because the auto in the function suggest to me, that also setters not only getters are proxied.

Simple example:
```Javascript
const mobx = require('mobx')

const observableObject = mobx.makeAutoObservable({
  a: 1,
  b: 2
})

mobx.autorun(() => {
  console.log(JSON.stringify(observableObject))
})

setTimeout(() => {
  observableObject.a = 2
}, 1500)
```

gives the right output, but also the strict mode warning:
```
{"a":1,"b":2}
[MobX] Since strict-mode is enabled, changing (observed) observable values without using an action is not allowed. Tried to modify: ObservableObject@1.a
{"a":2,"b":2}
```

If I replace the setting of the property `a` inside a `runInAction` call, than all is fine:
```Javascript
setTimeout(() => {
  mobx.runInAction(() => {
    observableObject.a = 2
  })
}, 1500)
```

```
{"a":1,"b":2}
{"a":2,"b":2}
```

How can I keep the simplicity of makeAutoObservable for normal Objects without  wrapping changes in runInAction?

**Comment by urugator** [comment, marked as answer]
The point of the warning is to avoid missing batch, consider:
```javascript
autorun(() => {
  console.log(observableObject.a, observableObject.b)
})

setTimeout(() => {
  observableObject.a = 2 // first autorun invocation
  observableObject.b = 2 // second autorun invocation
}, 1500)
// VS 
setTimeout(() => {
  runInAction(() => {    
    observableObject.a = 2
    observableObject.b = 2
  })  
  // one autorun invocation at the end of action (outermost if nested)
}, 1500)
```
There is no way for us to distinguish between setting one or multiple props. (Potentially is, but it's not reasonable)
If you don't like this behavior, you can turn it off with [`configure({ requireActions: "never" })`](https://mobx.js.org/configuration.html#enforceactions)
You're expected to expose behavior (actions) to consumers, rather than props.
